test/utils/clonableGenerator.js

```diff
@@ -0,0 +1,156 @@
+import test from 'tape';
+import { spy } from 'sinon';
+import { cloneableGenerator } from '../../src/internal/utils';
+
+test('it should proxy the original generator', assert => {
+  assert.plan(4);
+
+  const spyGenerator = {
```

Comment:
So I did a little digging to understand what is what.
According to the ECMA 2015 spec (just started to read it from scratch, very interesting stuff :) )
The correct naming would be:

`function* helloWorld() {...}`
=> helloWorld is a generator function, it is an instance of (GeneratorFunction) constructor and it extends the (Generator) prototype.

`const helloGen = helloWorld();`
=> helloGen is an instance of the helloWorld constructor, it extends helloGen.prototype which extends (Generator).prototype which extends (IteratorPrototype).

 helloGen is a kind of Generator, which is a kind of Iterator.

So I think it would be more accurate to name it gen for generator. Even if it seems confusing at first, I think it's better to stick as close as possible to the spec.

I find that [this schema](http://www.ecma-international.org/ecma-262/6.0/#sec-generatorfunction-objects) makes it more understandable.

Let me know if you think I missed/missunderstood sth :)